lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -0,0 +1,251 @@
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from pathlib import Path
+from ..command import Command
+from cortx.utils.conf_store import Conf
+from provisioner.commands import PillarSet
+from provisioner.salt import local_minion_id
+from .commons import Commons
+
+#TODO: Add this path in the global config
+prvsnr_cluster_path = Path(
+    '/opt/seagate/cortx_configs/provisioner_cluster.json'
+)
+
+
+"""Cortx Setup API for configuring the storage enclosure """
+
+
+class StorageEnclosureConfig(Command):
+    '''
+    $ cortx_setup storage config --name <enclosure-name>
+    $ cortx_setup storage config --type {RBOD|JBOD|EBOD|virtual}
+    $ cortx_setup storage config --user <user> --password <password>
+    $ cortx_setup storage config --controller_type {gallium|indium|virtual}
+    $ cortx_setup storage config --mode {primary|secondary} --ip <ip-address> --port <port-number>
+    '''
+    _args = {
+        'name': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'Set name of the enclosure e.g enc1-r10'
+        },
+        'type': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'Enclosure type e.g {RBOD | JBOD | EBOD | virtual}'
+        },
+        'user': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'username for controller e.g {manage}'
+        },
+        'password': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'password for controller user'
+        },
+        'controller_type': {
+            'type': str,
+            'optional': True,
+            'help': 'Type of controller e.g {gallium | indium | virtual}'
+        },
+        'mode': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'Controller mode e.g. {primary | secondary}'
+        },
+        'ip': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'IP address of the controller'
+        },
+        'port': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'Port of the controller to connect to'
+        }
+    }
+
+    def run(self, name=None, type=None,
```

Comment:
@mazinamdar , @83bhp 
Can this dest part help?
https://docs.python.org/3/library/argparse.html#the-add-argument-method
```dest - The name of the attribute to be added to the object returned by parse_args().```

Reply:
yes dest will help here ,I tried locally it will work